src/understand-tidb/the-lifecycle-of-a-statement.md

```diff
@@ -1 +1,67 @@
+		return cc.handleQuery(ctx, dataStr)
+```
+Where [mysql.ComQuery](https://github.com/pingcap/parser/blob/d4a88481405f8c59d45fc0a9c38ee24d55b9bf49/mysql/const.go#L102) is routed to [handleQuery](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/server/conn.go#L1633), which handles all different non-prepared statements (some commands like change database/schema or ping are handled directly in the dispatch function).
+
+## Statement string to Abstract Syntax Tree
+
+In [handleQuery](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/server/conn.go#L1633) we first needs to [parse](https://github.com/pingcap/parser/blob/d4a88481405f8c59d45fc0a9c38ee24d55b9bf49/yy_parser.go#L137) the statement string, which is done by [the parser, which is in its own repository](https://github.com/pingcap/parser) which is a MySQL compatible parser which parses statements and returns an Abstract Syntax Tree (AST) representing the statement.
+
+## AST -> Physical execution plan
+
+We now have a statement in AST form to handle in [handleStmt](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/server/conn.go#L1814)/[ExecuteStmt](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/session/session.go#L1620) where the Abstract Syntax Tree is [compiled](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/executor/compiler.go#L50) to a physical execution plan, including [optimizing](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/planner/optimize.go#L100) the execution plan, through a cost based optimizer. There are several steps in this process, such as name resolution, transaction management, [privilege checks](understand-tidb/privilege-management.md), handling given hints, etc.
+
+## Executing the optimized plan
+
+The optimized plan is executed through [runStmt](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/session/session.go#L1750), which builds an [executor](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/executor/adapter.go#L319) from the plan which will return a record set or directly execute the statements in case no records will be returned (like `INSERT`/`UPDATE`/`DELETE`). Before returning the record set, the executor starts the execution by calling the `Open()` API and the `Next()` api to retrieve the first chunk of data or execute the statement fully if no records are to be returned.
+
+### Requests sent to TiKV/TiFlash coprocessors
+
+### Sending the result back to the client
+
+If the statement returns a record set, it is handled in [writeChunks](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/server/conn.go#L1993) which loops over the record set's `Next()` 
+
+Notice that things like error handling, tracing etc. are not explained in this page.
+
+## TODO list (to be removed when completed)
+Agenda:
+
```

Comment:
Maybe move this to a task list in the description? 

https://docs.github.com/en/issues/tracking-your-work-with-issues/about-task-lists

Reply:
Thanks for the suggestion, it was some leftovers from the initial page suggestion, that I just kept as a TODO/work-in-progress. I just removed it since no longer needed.